src/coreclr/jit/promotion.cpp

```diff
@@ -2175,7 +2171,7 @@ bool ReplaceVisitor::ReplaceCallArgWithFieldList(GenTreeCall* call, GenTreeLclVa
     {
         Replacement* rep = nullptr;
         if (agg->OverlappingReplacements(argNode->GetLclOffs() + seg.Offset, seg.Size, &rep, nullptr) &&
-            rep->NeedsWriteBack)
+            !rep->NeedsReadBack)
```

Comment:
The previous check was conservative. It is possible for both `NeedsWriteBack` and `NeedsReadBack` to be false, in which case both the struct base local and replacement local are up to date; in that case we should prefer the value from the replacement local. This is the right check for that.